PWGHF/D2H/Tasks/taskDstarToD0Pi.cxx

```diff
@@ -139,67 +143,98 @@ struct HfTaskDstarToD0Pi {
     // Non Prmpt Gen
     registry.add("QA/hPtNonPromptDstarGen", "MC Matched Non-Prompt D* Candidates at Generator Level; #it{p}_{T} of  D*", {HistType::kTH1F, {{vecPtBins, "#it{p}_{T} (GeV/#it{c})"}}});
     registry.add("QA/hPtVsYNonPromptDstarGen", "MC Matched Non-Prompt D* Candidates at Generator Level; #it{p}_{T} of  D*; #it{y}", {HistType::kTH2F, {{vecPtBins, "#it{p}_{T} (GeV/#it{c})"}, {100, -5., 5.}}});
+
+    // Checking PV contributors from Data as well MC rec
+    registry.add("Efficiency/hNumPvContributorsAll", "PV Contributors; PV Contributor; FT0M Centrality", {HistType::kTH2F, {{100, 0, 300}, {axisCentrality}}}, true);
+    registry.add("Efficiency/hNumPvContributorsCand", "PV Contributors; PV Contributor; FT0M Centrality", {HistType::kTH2F, {{100, 0, 300}, {axisCentrality}}}, true);
+    registry.add("Efficiency/hNumPvContributorsCandInMass", "PV Contributors; PV Contributor; FT0M Centrality", {HistType::kTH2F, {{100, 0, 300}, {axisCentrality}}}, true);
   }
 
-  void process(CollisionsWCent const&, CandDstarWSelFlag const&)
+  void process(CollisionsWCent const& cols, soa::Filtered<CandDstarWSelFlag> const& selectedCands)
   {
-    for (const auto& candDstar : rowsSelectedCandDstar) {
-      auto yDstar = candDstar.y(constants::physics::MassDStar);
-      if (yCandDstarRecoMax >= 0. && std::abs(yDstar) > yCandDstarRecoMax) {
-        continue;
+    for (const auto& col : cols) {
+      auto nPVContributors = col.numContrib();
+      auto centrality = col.centFT0M();
+      registry.fill(HIST("Efficiency/hNumPvContributorsAll"), nPVContributors, centrality);
+
+      auto gIndexCol = col.globalIndex();
+      auto selectedCandsCurrentCol = selectedCands.sliceBy(preslicSelectedCandDstarPerCol, gIndexCol);
+      auto nCandsCurrentCol = selectedCandsCurrentCol.size();
+
+      if (nCandsCurrentCol > 0) {
+        LOGF(info, "size of selectedCandsCurrentCol: %d", nCandsCurrentCol);
```

Comment:
Do not put `LOGF(info)`, otherwise it will produce very large log files
```suggestion
        LOGF(debug, "size of selectedCandsCurrentCol: %d", nCandsCurrentCol);
```